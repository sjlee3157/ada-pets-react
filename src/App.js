import React, { Component } from 'react';
import axios from 'axios';

import PetList from './components/PetList';
import PetDetails from './components/PetDetails';
import SearchBar from './components/SearchBar';
import NewPetForm from './components/NewPetForm';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';

const URL = 'https://petdibs.herokuapp.com/pets';

// import pets from './data/pets.json'; // Removed for API data

class App extends Component {
  constructor(props) {
    super(props);
    // const petList = pets.map((pet) => {
    //   pet.currentPet = 0;
    //   pet.images = pet.images.map((filename) => {
    //     return `/images/${filename}`;
    //   });
    //   return pet;
    // });

    this.state = {
      petList: [],
      masterList: [],
      currentPet: undefined,
    };
  }

  componentDidMount() {

    axios.get(URL)
      .then((response) => {
        const pets = response.data.map((pet) => {
          console.log(pet.breed);
          const newPet = {
            ...pet,
            images: ['https://images.dog.ceo/breeds/germanshepherd/n02106662_10676.jpg'],
            species: pet.breed ? pet.breed.toLowerCase(): "",
            location: 'Seattle, WA',
            about: pet.about ? pet.about: "",
          };

          return newPet;
        }).filter((pet, index) =>  index < 10);

        this.setState({
          petList: pets,
          masterList: pets,
        });

      })
      .catch((error) => {
        console.log(error.message);
        this.setState({
          errorMessage: error.message,
        });

      });

  }

  onSelectPet = (petId) => {

    const selectedPet = this.state.petList.find((pet) => {
      return pet.id === petId;
    });
    if (selectedPet) {
      this.setState({
        currentPet: selectedPet,
      });
    }
  }

  onSearchChange = (value) => {
    console.log(value);
    const regex = new RegExp(`${value}`.toUpperCase());
    const petList = this.state.masterList.filter((pet) => {
      return regex.test(`${pet.name}${pet.about}${pet.species}`.toUpperCase());
    });

    this.setState({ petList: petList })

    if (currentPet && currentPet.id === id) {
      this.setState({ currentPet: undefined });
    }
  }

  addPet = (newPet) => {
    newPet.id = this.state.masterList.reduce((max = 0, currentPet) => max ? Math.max(max, currentPet.id): currentPet.id) + 1
    newPet.images = [newPet.image];
    const petList = [...this.state.masterList, newPet];

    this.setState({
      petList: petList,
      masterList: petList,
    });
  }

  removePet = (petId) => {
    let deleteIndex = -1;
    const pets = [...this.state.masterList];

    pets.forEach((pet, index) => {
      if (petId === pet.id) {
        deleteIndex = index;
      }
    }

    this.setState({
      petList: pets,
      masterList: pets,
    })
  }

      const matches = []
      for (let i = 0; i < pets.length; i++) {
        let pet = pets[i];
        let keywords = pet.name + ' ' + pet.species + ' ' + pet.about
        if (regex.test(keywords)) {
          matches.push(pet);
        };
      }
      this.setState({ filteredList: matches })
    }

    return (
      <main className="App">
        <header className="app-header">
          <h1>Ada Pets</h1>
          <h2>{this.state.errorMessage ? this.state.errorMessage: "" }</h2>
        </header>
        <section className="search-bar">
          <SearchBar
            searchPetsCallback={ searchPets }/>
        </section>
          { displayPetDetails() }
        <section className="pet-list">
          <PetList
            pets={ this.state.filteredList }
            onSelectPetCallback={ this.onSelectPetHandler }
            onRemovePetCallback={ this.onRemovePetHandler } />
        </section>
        <section className="new-pet-form-wrapper">
          <NewPetForm
            addPetCallback={ this.addPet } />
        </section>
      </main>
    );
  }
}

export default App;
