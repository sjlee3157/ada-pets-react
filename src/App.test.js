import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  // console.log(div.innerHTML);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches the snapshot', () => {
  const wrapper = shallow( <App />);
  expect(wrapper).toMatchSnapshot();
});

it('matches the deep snapshot', () => {
  const wrapper = mount( <App />);
  
  expect(wrapper).toMatchSnapshot();
  wrapper.unmount();
});