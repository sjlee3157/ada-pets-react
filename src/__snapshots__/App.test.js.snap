// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the deep snapshot 1`] = `
<App>
  <main
    className="App"
  >
    <header
      className="app-header"
    >
      <h1>
        Ada Pets
      </h1>
      <h2 />
    </header>
    <section
      className="search-bar"
    >
      <SearchBar
        onSearchChange={[Function]}
      >
        <section>
          <input
            className="search-bar"
            name="search-bar"
            onChange={[Function]}
            value=""
          />
        </section>
      </SearchBar>
    </section>
    <section
      className="pet-list"
    >
      <PetList
        deletePetCallback={[Function]}
        pets={Array []}
        selectPetCallback={[Function]}
      >
        <div
          className="card-group"
        />
      </PetList>
    </section>
    <section>
      <NewPetForm
        addPetCallback={[Function]}
      >
        <form
          className="new-pet-form"
          id="new-pet-form"
          name="new-pet-form"
          onSubmit={[Function]}
        >
          <div>
            <label
              className="new-pet-form--label"
              htmlFor="Name"
            >
              Name
            </label>
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              value=""
            />
          </div>
          <div>
            <label
              className="new-pet-form--label"
              htmlFor="species"
            >
              Species
            </label>
            <input
              name="species"
              onChange={[Function]}
              placeholder="species"
              value=""
            />
          </div>
          <div>
            <label
              className="new-pet-form--label"
              htmlFor="location"
            >
              Location
            </label>
            <input
              name="location"
              onChange={[Function]}
              placeholder="location"
              value=""
            />
          </div>
          <div>
            <label
              className="new-pet-form--label"
              htmlFor="image"
            >
              Image
            </label>
            <input
              name="image"
              onChange={[Function]}
              placeholder="image link"
              value=""
            />
          </div>
          <div>
            <label
              className="new-pet-form--label"
              htmlFor="about"
            >
              About
            </label>
          </div>
          <div>
            <textarea
              className="new-pet-form--about"
              name="about"
              onChange={[Function]}
              value=""
            />
          </div>
          <input
            className="btn btn-success new-pet-form--submit"
            name="submit"
            type="submit"
            value="Add a Pet"
          />
        </form>
      </NewPetForm>
    </section>
  </main>
</App>
`;

exports[`matches the snapshot 1`] = `
<main
  className="App"
>
  <header
    className="app-header"
  >
    <h1>
      Ada Pets
    </h1>
    <h2 />
  </header>
  <section
    className="search-bar"
  >
    <SearchBar
      onSearchChange={[Function]}
    />
  </section>
  <section
    className="pet-list"
  >
    <PetList
      deletePetCallback={[Function]}
      pets={Array []}
      selectPetCallback={[Function]}
    />
  </section>
  <section>
    <NewPetForm
      addPetCallback={[Function]}
    />
  </section>
</main>
`;
